redash/models.py

```diff
@@ -689,24 +581,45 @@ def should_schedule_next(previous_iteration, now, schedule):
     return now > next_iteration
 
 
-class Query(ChangeTrackingMixin, ModelTimestampsMixin, BaseVersionedModel, BelongsToOrgMixin):
-    id = peewee.PrimaryKeyField()
-    org = peewee.ForeignKeyField(Organization, related_name="queries")
-    data_source = peewee.ForeignKeyField(DataSource, null=True)
-    latest_query_data = peewee.ForeignKeyField(QueryResult, null=True)
-    name = peewee.CharField(max_length=255)
-    description = peewee.CharField(max_length=4096, null=True)
-    query = peewee.TextField()
-    query_hash = peewee.CharField(max_length=32)
-    api_key = peewee.CharField(max_length=40)
-    user = peewee.ForeignKeyField(User)
-    last_modified_by = peewee.ForeignKeyField(User, null=True, related_name="modified_queries")
-    is_archived = peewee.BooleanField(default=False, index=True)
-    schedule = peewee.CharField(max_length=10, null=True)
-    options = JSONField(default={})
-
-    class Meta:
-        db_table = 'queries'
+def generate_query_api_key(ctx):
+    return hashlib.sha1(u''.join((
+        str(time.time()), ctx.current_parameters['query'],
+        str(ctx.current_parameters['user_id']),
+        ctx.current_parameters['name'])).encode('utf-8')).hexdigest()
+
+def gen_query_hash(ctx):
+    return utils.gen_query_hash(ctx.current_parameters['query'])
+
+class Query(ChangeTrackingMixin, TimestampMixin, BelongsToOrgMixin, db.Model):
+    id = Column(db.Integer, primary_key=True)
+    version = Column(db.Integer)
+    org_id = Column(db.Integer, db.ForeignKey('organizations.id'))
+    org = db.relationship(Organization, backref="queries")
+    data_source_id = Column(db.Integer, db.ForeignKey("data_sources.id"), nullable=True)
+    data_source = db.relationship(DataSource)
+    latest_query_data_id = Column(db.Integer, db.ForeignKey("query_results.id"), nullable=True)
+    latest_query_data = db.relationship(QueryResult)
```

Comment:
Why do we need both `latest_query_data_id` and `latest_query_data`? (applies to all similar fields) SQLA doesn't have a convenience method to get the object id instead of loading the object itself otherwise?